Assignment 3-JS/script/script.js

```diff
@@ -0,0 +1,312 @@
+    }
+    return weeknum;
+  };
+  
+    var title = document.getElementById('modal-title');
+    var body = document.querySelector('.modal-body p');
+    var event_date = document.getElementById('event-date');
+    function triggerModal(obj) {
+      console.log(obj);
+      title.innerHTML = obj.title;
+      body.innerHTML = obj.discription;
+      event_date.innerHTML = obj.date;
+      $('#myModal').modal('show');
+    }
+  
+  //to open side-panel
+  document.getElementsByClassName('current-year-name')[0].addEventListener('click', () => {
+    document.getElementById("myScrollpanel").style.height = "120px";
+  });
+  
+  let arrow_up = document.getElementById('icon1');
+  
+  arrow_up.addEventListener('click',function() {
+    var scrollElement = document.querySelector('.firstElement-to-scroll');
+    var value = parseInt(scrollElement.innerHTML);
+    value = value + 10;
+    var year_names = document.getElementsByClassName('year-name');
+      for(let i = 0; i < year_names.length; i++) {
+        var check = year_names[i].innerHTML;
+        if(check == value){
+          scrollElement.classList.remove('firstElement-to-scroll');
+          year_names[i].classList.add('firstElement-to-scroll');
+          let nextScroll = document.querySelector('.firstElement-to-scroll');
+          nextScroll.scrollIntoView();
+        }
+        if (value > 2050) {
+          var scroll = year_names[0];
+          scroll.scrollIntoView();
+        }
+      }
+  });
+  
+  let arrow_down = document.getElementById('icon2');
+  
+  arrow_down.addEventListener('click',function(){
+    var scrollElement = document.querySelector('.firstElement-to-scroll');
+    var value = parseInt(scrollElement.innerHTML);
+    value = value - 10;
+    var year_names = document.getElementsByClassName('year-name');
+      for(let i = 0; i < year_names.length; i++) {
+        var check = year_names[i].innerHTML;
+        if(check == value){
+          scrollElement.classList.remove('firstElement-to-scroll');
+          year_names[i].classList.add('firstElement-to-scroll');
+          let nextScroll = document.querySelector('.firstElement-to-scroll');
+          nextScroll.scrollIntoView();
+        } if (value < 1950) {
+          var scroll = year_names[99];
+          scroll.scrollIntoView();
+        }
+      }
+  });
+  
+  const date = new Date();
+  const current_month = date.getMonth();
+  const current_year = date.getFullYear();
+  const current_date = date.getDate();
+  var month_days = document.getElementsByClassName('days')[0];
+  var weekDays = document.getElementById('week-number');
+  var monthArray = [
+          "january",
+          "february",
+          "march",
+          "april",
+          "may",
+          "june",
+          "july",
+          "august",
+          "september",
+          "october",
+          "november",
+          "december"
+      ];
+  
+  var events = {
+    "january-1": {
+      title: " Happy New year",

```

Comment:
What is the need of a space at the start of value?

Reply:
Removed spaces.